src/odemis/driver/test/technolution_test.py

```diff
@@ -446,30 +446,43 @@ def test_clock_VAs(self):
                 1 / clock_freq)
 
     def test_resolution_VA(self):
+        """
+        The setter allows only to enter resolutions which are a whole multiple of 4 times the mppc detector shape
+        """
         min_res = self.EBeamScanner.resolution.range[0][0]
         max_res = self.EBeamScanner.resolution.range[1][0]
 
         # Check if small resolution values are allowed
-        self.EBeamScanner.resolution.value = (min_res + 5, min_res + 5)
-        self.assertEqual(self.EBeamScanner.resolution.value, (min_res + 5, min_res + 5))
+        self.EBeamScanner.resolution.value = (self.MPPC.shape[0] * 4, self.MPPC.shape[0] * 4)
+        self.assertEqual(self.EBeamScanner.resolution.value, (self.MPPC.shape[0] * 4, self.MPPC.shape[0] * 4))
 
         # Check if big resolutions values are allowed
-        self.EBeamScanner.resolution.value = (max_res - 200, max_res - 200)
-        self.assertEqual(self.EBeamScanner.resolution.value, (max_res - 200, max_res - 200))
+        self.EBeamScanner.resolution.value = (max_res, max_res)
+        self.assertEqual(self.EBeamScanner.resolution.value, (max_res, max_res))
 
         # Check if VA refuses to set limits outside allowed range
         with self.assertRaises(IndexError):
             self.EBeamScanner.resolution.value = (max_res + 10, max_res + 10)
 
-        self.assertEqual(self.EBeamScanner.resolution.value, (max_res - 200, max_res - 200))
+        self.assertEqual(self.EBeamScanner.resolution.value, (max_res, max_res))
 
         with self.assertRaises(IndexError):
             self.EBeamScanner.resolution.value = (min_res - 1, min_res - 1)
-        self.assertEqual(self.EBeamScanner.resolution.value, (max_res - 200, max_res - 200))
+        self.assertEqual(self.EBeamScanner.resolution.value, (max_res, max_res))
 
         # Check if it is allowed to have non-square resolutions
-        self.EBeamScanner.resolution.value = (6000, 6500)
-        self.assertEqual(self.EBeamScanner.resolution.value, (6000, 6500))
+        self.EBeamScanner.resolution.value = (200 * self.MPPC.shape[0] * 4, 150 * self.MPPC.shape[0] * 4)
+        self.assertEqual(self.EBeamScanner.resolution.value, (200 * self.MPPC.shape[0] * 4, 150 * self.MPPC.shape[0] * 4))
+
+        # Check if values which are not a whole multiple of 4 times the mppc detector shape get changed correctly
+        self.EBeamScanner.resolution.value = (3207, 3207)
+        self.assertEqual(self.EBeamScanner.resolution.value, (3200, 3200))
+
+        self.EBeamScanner.resolution.value = (6403, 6403)
+        self.assertEqual(self.EBeamScanner.resolution.value, (6400, 6400))
+
+        self.EBeamScanner.resolution.value = (6416, 6416)
```

Comment:
Instead of having 2 tests, where the setter rounds down, also do one test where the setter rounds up.